sdk/iot/hub/samples/CMakeLists.txt

```diff
@@ -17,78 +17,95 @@ if(NOT OpenSSL_FOUND)
   find_package(OpenSSL REQUIRED)
 endif()
 
-set (sample_link_libraries 
-  az_iot_hub
-  az_iot_common
-  az_core
-  ${PAL}
-  eclipse-paho-mqtt-c::paho-mqtt3cs-static
-  OpenSSL::SSL
-  OpenSSL::Crypto
-)
-
 # C2D Sample
-add_executable (
-  paho_iot_hub_c2d_example 
+add_executable (paho_iot_hub_c2d_example
   ${CMAKE_CURRENT_LIST_DIR}/src/paho_iot_hub_c2d_example.c
-  )
+)
+
+target_link_libraries(paho_iot_hub_c2d_example
+  PRIVATE
+    az::core
+    az::iot::hub
+)
+
 
 target_link_libraries(
   paho_iot_hub_c2d_example PRIVATE
-  ${sample_link_libraries}
-)
+  eclipse-paho-mqtt-c::paho-mqtt3cs-static
+  OpenSSL::SSL
+  OpenSSL::Crypto
+  )
 
 # Methods Sample
-add_executable (
-  paho_iot_hub_methods_example 
+add_executable (paho_iot_hub_methods_example
   ${CMAKE_CURRENT_LIST_DIR}/src/paho_iot_hub_methods_example.c
-  )
+)
 
-target_link_libraries(
-  paho_iot_hub_methods_example PRIVATE
-  ${sample_link_libraries}
+target_link_libraries(paho_iot_hub_methods_example
+  PRIVATE
+    az::core
+    az::iot::hub
+)
+
+target_link_libraries(paho_iot_hub_methods_example
+  PRIVATE
+    eclipse-paho-mqtt-c::paho-mqtt3cs-static
+    OpenSSL::SSL
+    OpenSSL::Crypto
 )
 
 # Telemetry Sample
-add_executable (
-  paho_iot_hub_telemetry_example
+add_executable (paho_iot_hub_telemetry_example
   ${CMAKE_CURRENT_LIST_DIR}/src/paho_iot_hub_telemetry_example.c
-  )
+)
 
-target_link_libraries(
-  paho_iot_hub_telemetry_example PRIVATE
-  ${sample_link_libraries}
+target_link_libraries(paho_iot_hub_telemetry_example
+  PRIVATE
```

Comment:
nit: Add a comment explaining why these are private rather than public (samples are executable, so we don't need to declare the dependencies publicly).

Reply:
Just to keep comments a little lower, I'll probably leave as is since this is a common cmake pattern